### ISSUE TITLE
Check field computations

### ISSUE THREAD

**Issue by momchilmm** [fancompute/legume#24]
Would be good to check the computed fields of the eigenmodes vs. some first-principle simulation.

Note that the H-field should be the most "correct" as all components are continuous across all interfaces. The D and especially the E fields, because of the way the expansion works, will have some spurious discontinuities even in cases when they're not supposed to. For example if a patterned and un-patterned slab of the same material sit on top of each other, there is no "interface" in the region where they have the same permittivity, but because the effective permittivity of the patterned slab is lower, there's a discontinuity in the guided modes, and so there will be one in the PhC modes too.

**Comment by abouscal** [comment]
Hello,
I noticed indeed that for some geometries, the E field can show discontinuities that are not present when doing a complete FDTD calculation. Is there any progress on this issue ? Could it be possible to determine the E field from H directly or would we lose the speed up allowed by GME? I'm new to these kind of numerical methods so I'm not sure how E is derived. 
![image](https://user-images.githubusercontent.com/70629839/96278511-dfac6500-0fd5-11eb-8a3d-af0acebc6265.png)
Regards

**Comment by momchilmm** [comment]
Yeah so specifically the z-dependence of the permittivity is not captured well in the E-field computation, and so it causes a discontinuity in a place where there is no interface. This is in particular a problem for field localized in air, as you can see above. I think it could be worth trying to compute E from H(r) and eps(r) by just doing the curl numerically in real space, but I don't have time to experiment with this right now. However, if you do try it out and it looks good compared to other solvers, I will work on adding it to legume.
